src/database/DBOperation.cpp

```diff
@@ -293,6 +295,42 @@ bool DBOperationGet::verify_class(const dclass::Class *dclass)
     return true;
 }
 
+bool DBOperationGet::is_independent_of(const DBOperation *other) const
+{
+    if(doid() != other->doid()) {
+        // Not even operating on the same doId; we're good.
+        return true;
+    }
+
+    switch(other->type()) {
+        case GET_OBJECT:
+        case GET_FIELDS:
+            return true;
+
+        case SET_FIELDS:
+        case UPDATE_FIELDS:
+            if(type() == GET_OBJECT) {
+                // A GET_OBJECT gets everything, and is therefore not
+                // independent of any changes to the object.
+                return false;
+            }
+
+            // Test if any of the fields we're getting is a field changed by
+            // the other operation...
```

Comment:
Each DBOperationXYZ class has its own logic. The SET_FIELDS->SET_FIELDS is lower in the file. (The function you're looking at is only for determining GET vs. everything else.)